napari/layers/base/base.py

```diff
@@ -735,67 +722,62 @@ def refresh(self, event=None):
             self.set_view_slice()
             self.events.set_data()
             self._update_thumbnail()
-            self._update_coordinates()
+            self._update_value_and_status()
             self._set_highlight(force=True)
 
-    def _update_coordinates(self):
-        """Insert the cursor position into the correct position in the
-        tuple of indices and update the cursor coordinates.
-        """
-        coords = list(self.coordinates)
-        for d, p in zip(self.dims.displayed, self.position):
-            coords[d] = p
-        self.coordinates = tuple(coords)
+    @property
+    def coordinates(self):
+        """Cursor position in data coordinates."""
+        return tuple(self._transforms[1:].simplified.inverse(self.position))
```

Comment:
What is 1: for here? Maybe add a comment to explain it?

Also, is simplified cached? If not, it's probably not worth accessing it over and over.

Reply:
Simplified is not cached, but that would be nice to do. Maybe something i can get help with in a follow-up PR!!

Reply:
comment added